DemoViper/VIPER/Module.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  Module.swift
+//  viper
+//
+//  Created by Su Van Ho on 9/8/19.
+//  Copyright © e2019 Nimble Co. Ltd. All rights reserved.
+//
+
+import Foundation
+
+/// These protocols is defined for people who want to take a look on a big picture of VIPER, optional for using.
+///
+/// Module is used for wrap your module, it contains View, Presenter, Router, Interactor and connect all parts of VIPER together.
+protocol Module {
```

Comment:
`Module` should also conform to `AnyObject` protocol. 

because module is holding VIPER component's properties which are reference type. 

![picture](https://i.imgur.com/3JFsgDD.png)

Reply:
https://compass.nimblehq.co/development/code-conventions/ios/

Reply:
Fixed